### PR TITLE
fix(web): default parse of length styling for empty string

### LINKED ISSUE #11234: bug(web): empty-string style parse lacks proper default
Sentry Issue: [KEYMANWEB-COM-1HG](https://keyman.sentry.io/issues/5198209062/?referrer=github_integration)

```
TypeError: null is not an object (evaluating 'e.val')
  at new ParsedLengthStyle (https://s.keyman.com/kmw/engine/17.0.308/src/web/src/engine/osk/src/lengthStyle.ts:20:16)
  at _VisualKeyboard.constructLayoutParams (https://s.keyman.com/kmw/engine/17.0.308/src/web/src/engine/osk/src/visualKeyboard.ts:1305:27)
  at _VisualKeyboard.refreshLayout (https://s.keyman.com/kmw/engine/17.0.308/src/web/src/engine/osk/src/visualKeyboard.ts:1295:40)
  at <anonymous> (https://s.keyman.com/kmw/engine/17.0.308/src/web/src/engine/osk/src/visualKeyboard.ts:1387:12)
```

### PR DESCRIPTION
**Pull request by jahorton**
Fixes #11234.

@keymanapp-test-bot skip

### CODE REVIEW

File: web/src/engine/osk/src/lengthStyle.ts [hunk 1]
```diff
@@ -61,12 +61,15 @@ export class ParsedLengthStyle implements LengthStyle {
...
   private static parseLengthStyle(spec: string): LengthStyle {
+    if(spec == '') {
+      return { val: 1, absolute: false };
+    }
+
     const val = parseFloat(spec);
 
     if(isNaN(val)) {
       // Cannot parse.
       console.error("Could not properly parse specified length style info: '" + spec + "'.");
```
Comment (mcdurdin) [inline web/src/engine/osk/src/lengthStyle.ts:72]: Given val is NaN here, what should we be returning?

```suggestion
    if(isNaN(val)) {
      // Cannot parse.
      console.error("Could not properly parse specified length style info: '" + spec + "'.");
      return { val: 1, absolute: false };
```

File: web/src/test/auto/headless/engine/osk/lengthStyles.js [hunk 2]
```diff
@@ -4,6 +4,12 @@ import sinon from 'sinon';
 import { ParsedLengthStyle } from 'keyman/engine/osk';
 
 describe('Length style processing', () => {
+  it("constructs from empty-string", () => {
+    let length = new ParsedLengthStyle('');
+    assert.equal(length.val, 1);
+    assert.equal(length.absolute, false);
+  })
+
```
Comment (mcdurdin) [inline web/src/test/auto/headless/engine/osk/lengthStyles.js:12]:
```suggestion

  it("constructs from garbage", () => {
    let length = new ParsedLengthStyle('garbage');
    assert.equal(length.val, 1);
    assert.equal(length.absolute, false);
  })

```

**Review by mcdurdin — APPROVED** [review]
Clearly returning null is verboten so for robustness, should we still return something valid if input is garbage?

Otherwise, LGTM

File: web/src/engine/osk/src/lengthStyle.ts (hunk 1 above)
Comment (jahorton) [reply web/src/engine/osk/src/lengthStyle.ts:72]: I actually did this exact suggestion at first, but figured that if we ever run into a spot with an actual garbage string in production, having a clear error would be wise.  I don't see any Sentry events for the error string, and that's a bit concerning.

**Review by ermshiperete — APPROVED** [review]

**Comment by keyman-server** [comment]
Changes in this pull request will be available for download in [Keyman version 17.0.311-beta](https://keyman.com/downloads/releases/beta/17.0.311)

**Pull request merged**